### PR TITLE
Various internal improvements

### LINKED ISSUE #340: Raphael getBBox
@neveldo 

I recently created the function `getBBoxWithCenter()` which get the Boundary Box of an element, and add its center coordinates cx, cy.

However, I noticed that this information is already here!

With Raphael 2.2.7 (latest) : http://jsfiddle.net/can4e7q0/1/
`element.getBBox()` will return the following object:
```
 { x, y, x2, y2, width, height, cx, cy, toString }
```
However, if I go back to Raphael 2.1.0, jsfiddle.net/can4e7q0/1 it will return the following object:
```
 { x, y, x2, y2, width, height, toString }
```

Indeed, after looking through Raphael commits, I found this one: https://github.com/DmitryBaranovskiy/raphael/commit/2dd068a94c123ff10deb930aa48d279896b45864

So, my question is: should we ditch the internal function and use the getBBox which give all we need? However, I fear some people are using older version of Raphael and thus may encounter issues. 

In `package.json`, we ask for a minimum of Raphael 2.1.0. If we bump it to 2.1.1, we can get the complete BBox values.

Or we can keep the internal function and check if cx/cy are defined and add them if needed.

What do you think?

### PR DESCRIPTION
**Pull request by Indigo744**
This PR implements some various internal improvements.

The main enhancement concerns the option: each element option is stored in the element object, alongside mapElem and textElem. This allows to simplify quite a bit of various other stuff related to the options, like HTML links options, tooltip options, event handlers...

Also, the getBBoxCenter is removed (fix #340).

### CODE REVIEW

File: js/jquery.mapael.js [hunk 1]
```diff
@@ -1469,6 +1467,18 @@
                 elem.mapElem.toFront();
             }
 
+            // Set the cursor attribute related to the HTML link
+            if (elem.options.href !== undefined) {
+                elem.options.attrs.cursor = "pointer";
+                if (elem.options.text) elem.options.text.attrs.cursor = "pointer";
+            } else {
+                // No HTML links, check if a cursor was defined to pointer
+                if (elem.mapElem.attrs.cursor === 'pointer') {
```
Comment (neveldo) [inline js/jquery.mapael.js]: Why did you add this check before setting the cursor style to 'auto' ?

Comment (Indigo744) [reply js/jquery.mapael.js]: In case the user did set the cursor to something other than pointer (like `help`...)...

Comment (Indigo744) [reply js/jquery.mapael.js]: In order to avoid overriding the user attribute.

**Comment by neveldo** [comment]
Storing the area / plot options within the objects next to mapElem and textElem is a great idea that allowed you to introduce a lot of simplifications , the elements configuration parts are a lot more clear now. I'm also glad to see the code factorization within the two new functions that you have introduced  (setPlotAttributes and setPlotCoords).

File: js/jquery.mapael.js (hunk 1 above)
Comment (neveldo) [reply js/jquery.mapael.js]: Ok thanks, I got it ! It will not work for the users who willingly set cursor to 'pointer', but it should be a very unusual case, and its still better than before :)
